### DISCUSSION TITLE
[JOB OFFER] $150 Pool Object Distance Module

### DISCUSSION THREAD

**Discussion by heyaapl** [azerothcore/azerothcore-wotlk#3942] [Job offers]
I'm looking to have a module that allows for a minimum distance between GOB Pool Spawns. I've been working for the better part of four months cleaning up all GOB nodes (Ore, Herbs, Chests, Quest Objects, etc) to reduce the over-spawning issues by remapping the entire GOB Pool tables. However, I'm now running into issues with how AzerothCore spawns in new nodes.

At a high-level, I'm thinking this would require either an additional table or single column to the existing Pool_Template table that would store a minimum distance that must exist between node spawning. A default value of 0 keeps the current spawning system the same, while a value of say 15 would require 15 points would require a distance of 15 XYZ coordinates is required before a node could spawn next to it. This will help (a) reduce the number of groupings needed in the database to better control spawn rates of nodes (b) allow for easier maintenance and control for server owners who would like to control the spawn rates of nodes.

After you develop this, I'd like to have it made available as a free module for all AzerothCore users to benefit from. 

The bounty for this Module is $150.

**Comment by FrancescoBorzi** [comment]
If a rewrite/improvement/etc... of the node spawning system is needed, I think it would be better to do it directly in the core (and move this post as a Bounty in the main repo).

However, I've never investigated it so I personally can't tell much. Any feedback is welcome here.

**Comment by solidmaxtor** [comment]
Decide better way if modular or directly in core, like this heyaapl i am colaborate for this change. <3

**Comment by heyaapl** [comment]
@solidmaxtor Awesome - I'm not certain where this should go, but I put opened an issue and placed a Bounty against it. A module might be easier, but considering it will probably require some Core changes, idk if that can go in as a module.
